src/roboteq_controller.cpp

```diff
@@ -0,0 +1,13 @@
+
+namespace roboteq {
+
+void roboteq_controller::roboteq_controller(std::unique_ptr<i_comm> &&comm) : _comm(comm){
```

Comment:
my recollection of move semantics is not the best but i'm pretty sure this is how this works. You pass in a rvalue reference to this constructor using `std::move` on the lvalue as seen in the `example_main.cpp` and then when u pass `comm`to `_comm` in the initializer list, `comm` should be move assigned to `_comm`. if you have any trouble with these lines, lemme know.